docs/src/tutorial/pxd_files.rst

```diff
@@ -1,48 +1,81 @@
 .. _pxd_files:
 
+*********
 pxd files
-=========
+*********
 
-In addition to the ``.pyx`` source files, Cython uses ``.pxd`` files
+.. include::
+    ../two-syntax-variants-used
+
+In addition to the ``.pyx`` and ``.py`` source files, Cython uses ``.pxd`` files
 which work like C header files -- they contain Cython declarations
 (and sometimes code sections) which are only meant for inclusion by
-Cython modules.  A ``pxd`` file is imported into a ``pyx`` module by
+Cython modules.  A ``.pxd`` file is imported into a ``.pyx`` module by
 using the ``cimport`` keyword.
 
-``pxd`` files have many use-cases:
+``.pxd`` files have many use-cases:
 
 1.  They can be used for sharing external C declarations.
 2.  They can contain functions which are well suited for inlining by
-    the C compiler. Such functions should be marked ``inline``, example::
+    the C compiler. Such functions should be marked ``inline``, example:
+
+    .. tabs::
+
+        .. group-tab:: Pure Python
+
+            .. literalinclude:: ../../examples/tutorial/pxd_files/inline.py
+
+        .. group-tab:: Cython
 
-       cdef inline int int_min(int a, int b):
-           return b if b < a else a
+            .. literalinclude:: ../../examples/tutorial/pxd_files/inline.pyx
 
-3.  When accompanying an equally named ``pyx`` file, they
+3.  When accompanying an equally named ``.pyx`` / ``.py`` file, they
     provide a Cython interface to the Cython module so that other
     Cython modules can communicate with it using a more efficient
     protocol than the Python one.
 
-In our integration example, we might break it up into ``pxd`` files like this:
+In our integration example, we might break it up into ``.pxd`` files like this:
+
+1.  Add a ``cmath.pxd``:
+
+    .. literalinclude:: ../../examples/tutorial/pxd_files/cmath.pxd
+        :caption: cmath.pxd
+
+    Then one would simply do
+
+    .. tabs::
+
+        .. group-tab:: Pure Python
+
+            .. literalinclude:: ../../examples/tutorial/pxd_files/integrate.py
+                :caption: integrate.py
+
+            .. include::
+                ../cimport-warning
+
+        .. group-tab:: Cython
+
+            .. literalinclude:: ../../examples/tutorial/pxd_files/integrate.pyx
+                :caption: integrate.pyx
```

Comment:
`integrate` certainly isn't the right name for these files.
And they don't feel relevant enough to be in files at all. There's not much to test about them.

Reply:
Well, to be honest, I didn't come up with it, it was there, I just adapted it. If you look [here](https://github.com/cython/cython/pull/4295/files#diff-3c94f63d823eb89992e39817d55f31450fc8a119789a3415c989fe19eedf6e21L30):
> Then one would simply do ``from cmath cimport sin`` in ``integrate.pyx``.

you can see that this code is referred to as being in a file.

Of course, we can redo all of this in a better way.

Reply:
> `integrate` certainly isn't the right name for these files.

How should we name it then?

Reply:
I looked into that .. and actually, the reason for this code to be in the file is alignment, you can not align with empty lines inside of the `code-block`, so I'd say it's not that bad to add files for nicer looks, what do you think @scoder?

Reply:
> you can see that this code is referred to as being in a file.

Ah, ok, sorry. Then I didn't read the text well enough here.

> > `integrate` certainly isn't the right name for these files.
> 
> How should we name them then?

Maybe just `use_cmath_sin.pyx` ?

Reply:
Indeed this example is confusing, the pyx and pxd are completely unrelated as it seems

![image](https://github.com/cython/cython/assets/52697657/2179b9b7-92bf-4cf8-bc9a-b401837ff238)

Reply:
Let's keep the names as they are.